db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/NumericTypeCompiler.java

```diff
@@ -364,6 +364,53 @@ else if (typeId.isBigIntTypeId() &&
 			}
 			else if (typeId.isDecimalTypeId())
 			{
+				/* DB-9425
+				 * Make sure we still have enough digits for the integral part.
+				 * Reduce scale when necessary, or leave it when not possible.
+				 */
+				{
+					int lprec = leftType.getPrecision();
+					int lscale = leftType.getScale();
+					int rprec = rightType.getPrecision();
+					int rscale = rightType.getScale();
+					int integralNumDigits;
+
+					switch (operator) {
+						case TypeCompiler.SUM_OP:
+						case TypeCompiler.MINUS_OP:
```

Comment:
I think PLUS_OP should be added here similar to MINUS_OP.

Reply:
Yes, it was a mistake. It should be PLUS_OP, not SUM_OP.